packages/devtools_app/lib/src/code_size/code_size_processor.dart

```diff
@@ -0,0 +1,95 @@
+// Copyright 2020 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:convert';
+
+import '../charts/treemap.dart';
+import 'canned_data.dart';
+import 'code_size_controller.dart';
+
+class CodeSizeProcessor {
```

Comment:
The processor / data model should not reach back to the controller. 
it looks like you can make loadJson return a TreemapNode to avoid a dependency on the controller. That will keep the layering of functionality a bit simpler and let you test the processor if you wanted to without depending on the controller which would be nice if you wanted to add some basic unittests of it.